sentry-android-core/src/test/java/io/sentry/android/core/ConnectivityCheckerTest.kt

```diff
@@ -1,31 +1,161 @@
+        val cm = mock<ConnectivityManager>()
+        whenever(contextMock.getSystemService(any())).thenReturn(cm)
+        val an = mock<NetworkInfo>()
+        whenever(cm.activeNetworkInfo).thenReturn(an)
+        whenever(an.isConnected).thenReturn(true)
+        assertTrue(ConnectivityChecker.isConnected(contextMock, mock())!!)
+    }
+
+    @Test
+    fun `When network is active but not connected, return false for isConnected`() {
+        val contextMock = mock<Context>()
+        val cm = mock<ConnectivityManager>()
+        whenever(contextMock.getSystemService(any())).thenReturn(cm)
+        val an = mock<NetworkInfo>()
+        whenever(cm.activeNetworkInfo).thenReturn(an)
+        whenever(an.isConnected).thenReturn(false)
+        assertFalse(ConnectivityChecker.isConnected(contextMock, mock())!!)
+    }
+
+    @Test
+    fun `When there's no permission, return true for isConnected`() {
+        val contextMock = mock<Context>()
+        val cm = mock<ConnectivityManager>()
+        whenever(contextMock.getSystemService(any())).thenReturn(cm)
+        whenever(contextMock.checkPermission(any(), any(), any())).thenReturn(PERMISSION_DENIED)
+        assertTrue(ConnectivityChecker.isConnected(contextMock, mock())!!)
+    }
+
+    @Test
+    fun `When network is not active, return false for isConnected`() {
+        val contextMock = mock<Context>()
+        val cm = mock<ConnectivityManager>()
+        whenever(contextMock.getSystemService(any())).thenReturn(cm)
+        assertFalse(ConnectivityChecker.isConnected(contextMock, mock())!!)
+    }
+
+    @Test
+    fun `When ConnectivityManager is not available, return null for isConnected`() {
+        assertNull(ConnectivityChecker.isConnected(mock(), mock()))
+    }
+
+    @Test
+    fun `When ConnectivityManager is not available, return null for getConnectionType`() {
+        val buildInfo = mock<IBuildInfoProvider>()
+        whenever(buildInfo.sdkInfoVersion).thenReturn(Build.VERSION_CODES.M)
+        assertNull(ConnectivityChecker.getConnectionType(mock(), mock(), buildInfo))
+    }
+
+    @Test
+    fun `When sdkInfoVersion is not min Marshmallow, return null for getConnectionType`() {
+        val buildInfo = mock<IBuildInfoProvider>()
+        whenever(buildInfo.sdkInfoVersion).thenReturn(0)
+        assertNull(ConnectivityChecker.getConnectionType(mock(), mock(), buildInfo))
+    }
+
+    @Test
+    fun `When there's no permission, return null for getConnectionType`() {
+        val contextMock = mock<Context>()
+        val cm = mock<ConnectivityManager>()
+        whenever(contextMock.getSystemService(any())).thenReturn(cm)
+        whenever(contextMock.checkPermission(any(), any(), any())).thenReturn(PERMISSION_DENIED)
+        val buildInfo = mock<IBuildInfoProvider>()
+        whenever(buildInfo.sdkInfoVersion).thenReturn(Build.VERSION_CODES.M)
+        assertNull(ConnectivityChecker.getConnectionType(contextMock, mock(), buildInfo))
```

Comment:
`m`: What about return an empty string or`"unknown"` and/or also defining constants for `"wifi"`, `"ethernet"`, `"cellular"`, and `"unknown"`?

Reply:
we'll be serializing and sending over the network something useless IMO.
if you read a crash report and see `network type: unknown`, is this any useful?

Reply:
Now we send `network type: null`. With this I might ask myself why is it `null`? Is it working properly or do we not know. Sending `network type: unknown` is clearer IMO because it says we can't tell.

Reply:
@philipphofmann we are not sending it `null`, we are not sending it at all, `null` values are omitted by default on serialization.
`Gson` needs `serializeNulls()` to serialize null values.